backend/connector-integration/src/connectors/checkout/transformers.rs

```diff
@@ -0,0 +1,773 @@
+            response,
+            router_data,
+            http_code,
+        } = item;
+
+        // Get attempt status from payment status
+        let status = enums::AttemptStatus::from(response.status);
+
+        let mut router_data = router_data;
+        router_data.resource_common_data.status = status;
+
+        // Check if the response indicates an error
+        if status == enums::AttemptStatus::Failure {
+            router_data.response = Err(ErrorResponse {
+                status_code: http_code,
+                code: response
+                    .response_code
+                    .unwrap_or_else(|| NO_ERROR_CODE.to_string()),
+                message: response
+                    .response_summary
+                    .clone()
+                    .unwrap_or_else(|| NO_ERROR_MESSAGE.to_string()),
+                reason: response.response_summary,
+                attempt_status: None,
+                connector_transaction_id: Some(response.id.clone()),
+            });
+        } else {
+            // Handle successful response
+            router_data.response = Ok(PaymentsResponseData::TransactionResponse {
+                resource_id: ResponseId::ConnectorTransactionId(response.id.clone()),
+                redirection_data: Box::new(None),
+                mandate_reference: Box::new(None),
+                connector_metadata: None,
+                network_txn_id: None,
+                connector_response_reference_id: Some(response.reference.unwrap_or(response.id)),
+                incremental_authorization_allowed: None,
+                raw_connector_response: None,
+            });
+        }
+
+        Ok(router_data)
+    }
+}
+
+// Implementation for PaymentCaptureRequest
+impl
+    TryFrom<
+        super::CheckoutRouterData<
+            RouterDataV2<Capture, PaymentFlowData, PaymentsCaptureData, PaymentsResponseData>,
+        >,
+    > for PaymentCaptureRequest
+{
+    type Error = error_stack::Report<ConnectorError>;
+    fn try_from(
+        item: super::CheckoutRouterData<
+            RouterDataV2<Capture, PaymentFlowData, PaymentsCaptureData, PaymentsResponseData>,
+        >,
+    ) -> Result<Self, Self::Error> {
+        let router_data = &item.router_data;
+        let connector_auth = &router_data.connector_auth_type;
+        let auth_type: CheckoutAuthType = connector_auth.try_into()?;
+        let processing_channel_id = auth_type.processing_channel_id;
+
+        // Determine if this is a multiple capture by checking if multiple_capture_data exists
+        let capture_type = if router_data.request.multiple_capture_data.is_some() {
+            CaptureType::NonFinal
+        } else {
+            CaptureType::Final
+        };
+
+        // Get optional reference for multiple captures
+        let reference = router_data
+            .request
+            .multiple_capture_data
+            .as_ref()
+            .map(|mcd| mcd.capture_reference.clone());
```

Comment:
```suggestion
            .map(|multiple_capture_data| multiple_capture_data.capture_reference.clone());
```

Use proper variable naming